std/typecons.d

```diff
@@ -3706,7 +3741,58 @@ auto nullableRef(T)(T* t)
     NullableRef!TestToString ntts = &tts;
     assert(ntts.to!string() == "2.5");
 }
+@safe unittest
+{
+    enum Foo {a, b, c}
+    assert(Nullable!Foo.sizeof == Foo.sizeof);
+
+    {
+        auto foo = nullable(Foo.a);
+        assert(foo.sizeof == Foo.sizeof);
+        assert(!foo.isNull);
+        assert(foo == Foo.a);
+        assert(foo.get == Foo.a);
+        foo = Foo.b;
+        assert(!foo.isNull);
+        assert(foo == Foo.b);
+        foo.nullify();
+        assert(foo.isNull);
+    }
+    {
+        auto foo = Nullable!Foo.init;
+        assert(foo.sizeof == Foo.sizeof);
+        assert(foo.isNull);
+        foo = Foo.c;
+        assert(!foo.isNull);
+        assert(foo == Foo.c);
+    }
 
+    enum Foo2 : int {a = int.min}
+    assert(Nullable!Foo2.sizeof == Foo2.sizeof);
+
+    enum Foo3 : int {a = int.min, b = int.max}
+    assert(Nullable!Foo3.sizeof != Foo3.sizeof);
+}
+@safe unittest
+{
+    class FooClass { }
+    assert(Nullable!FooClass.sizeof == FooClass.sizeof);
+    class FooInterface { }
+    assert(Nullable!FooInterface.sizeof == FooInterface.sizeof);
+}
+@safe unittest
```

Comment:
What's the purpose of this test?

Reply:
Because `nullValue == false` when the base type is bool, I think he wanted to ensure that the code won't consider non-null `false` being null, or something like that. I think that's a good rationale.